### PR TITLE
 feat: Add capability to use messaging based Command Client

### PR DESCRIPTION
**Pull request by lenny-intel**
If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-bootstrap/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?) Full testing will be done in App SDK PR
- [x] I have opened a PR for the related docs change (if not, why?)
  TBD

## Testing Instructions
N/A

### CODE REVIEW

**Comment by lenny-intel** [comment]
@cloudxxx8 , If you merge, please `Sqaush and Merge` and remove the first commit message. Not sure how that got there, but the list of changed files is correct. THX!

**Review by hahattan — CHANGES_REQUESTED** [review]

File: bootstrap/handlers/clients.go [hunk 1]
```diff
@@ -85,9 +93,34 @@ func (cb *ClientsBootstrap) BootstrapHandler(
 			})
 
 		case common.CoreCommandServiceKey:
+			var client interfaces.CommandClient
+
+			if serviceInfo.UseMessageBus {
+				// TODO: Move following outside loop when multiple messaging based clients exist
+				messageClient := container.MessagingClientFrom(dic.Get)
+				if messageClient == nil {
+					lc.Errorf("Unable to create Command client using MessageBus: %s", "MessageBus Client was not created")
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply bootstrap/handlers/clients.go:102]: I did this on purpose so tests can verify Errorf called.

File: bootstrap/handlers/clients.go [hunk 2]
```diff
@@ -85,9 +93,34 @@ func (cb *ClientsBootstrap) BootstrapHandler(
...
+
+				// TODO: Move following outside loop when multiple messaging based clients exist
+				timeout, err := time.ParseDuration(config.GetBootstrap().Service.RequestTimeout)
+				if err != nil {
+					lc.Errorf("Unable to parse Service.RequestTimeout as a time duration: %v", err)
+					return false
+				}
+
+				// Note that NewCommandClient never returns an error
+				client, _ = clientsMessaging.NewCommandClient(messageClient, serviceInfo.Topics, timeout)
```
[the remark this answers is not in the document]
Comment (lenny-intel) [reply bootstrap/handlers/clients.go:114]: Yep, missed that. THX! Will add err check.

[the remark this answers is not in the document]
Comment (lenny-intel) [reply bootstrap/handlers/clients.go]: Fixed

**Review by hahattan — APPROVED** [review]
LGTM

**Comment by lenny-intel** [comment]
@cloudxxx8 , need your approval as committer

**Review by jim-wang-intel — APPROVED** [review]

**Pull request merged**